### PR TITLE
created interactive cli for private key connect

### PR DESCRIPTION
**Pull request by ghost**
- changed interactive prompts for better clarity
- changed prompt styling and colors for better readability
- fixed checksum error
- refactored code 
- added support for more options for pvt key connect

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -101,4 +101,6 @@ ctx*.json
 
 # fordefi
 private.pem
-token
\ No newline at end of file
+token
+
+--get-hacked
```
Comment (avinash-onebyte) [inline .gitignore:106]: what is this for ?

File: src/cli.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Command, OptionValues } from 'commander'
 import { UnsignedTxJson, SignedTxJson, Context, ContextFile, FlareTxParams } from './interfaces'
-import { contextEnv, contextFile, getContext } from './constants'
+import { contextEnv, contextFile, getContext,colorCodes,networkMapping } from './constants'
```
Comment (avinash-onebyte) [inline src/cli.ts]: No spaces -> use prettier extension to fix these formatting errors automatically.

File: src/cli.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ import { ledgerSign, signId } from './ledger/sign'
 import { getSignature, sendToForDefi } from './forDefi'
 import { createWithdrawalTransaction, sendSignedWithdrawalTransaction } from './withdrawal'
 import { log, logError, logInfo, logSuccess } from './output'
+const prompts = require('./prompts');
```
Comment (avinash-onebyte) [inline src/cli.ts]: Use import instead of require

File: src/cli.ts [hunk 4]
```diff
@@ -56,6 +57,31 @@ export async function cli(program: Command) {
...
       }
     })
+  //interactive cli
+  program
+    .command("interactive").description("interactive cli")
+    .action(async () => {
+      try {
+        const {wallet,path} =  await connectWallet()
+        const network = await selectNetwork()
+        const task = await selectTask()
```
Comment (avinash-onebyte) [inline src/cli.ts:68]: Why so many synchronous functions usage here, use Promise.all here as all these async code don't have interdependency

File: src/cli.ts [hunk 5]
```diff
@@ -56,6 +57,31 @@ export async function cli(program: Command) {
...
+        const options = getOptionsInterativeCli(program,wallet,path,network)
+        const ctx = await contextFromOptions(options)
+        if (task == 'View chain addresses') {
+          logAddressInfo(ctx)
+        } else if (task == 'Check on-chain balance') {
```
Comment (avinash-onebyte) [inline src/cli.ts]: create a file screenConstants and move all the strings over there, so that devs don't have to go through code to change the text

**Review by avinash-onebyte — COMMENTED** [review]
Added some comments

File: .gitignore (hunk 1 above)
Comment (ghost) [reply .gitignore:106]: this file is created when pvt key is used, added to gitignore to keep it out of the repository

File: src/screenConstants.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const screenConstants = {
+    viewChainAddress: 'View chain addresses',
```
Comment (avinash-onebyte) [inline src/screenConstants.ts]: Use ALL CAPS for string components for key names

**Review by avinash-onebyte — COMMENTED** [review]
minor change

**Review by avinash-onebyte — APPROVED** [review]

**Pull request merged**